continuous_integration/install.sh

```diff
@@ -47,7 +47,15 @@ if [[ "$DISTRIB" == "conda" ]]; then
 
 elif [[ "$DISTRIB" == "ubuntu" ]]; then
     # Use standard ubuntu packages in their default version
-    sudo apt-get install -qq python-scipy python-nose python-pip
+    sudo apt-get install -qq python-scipy
+    # At the time of writing numpy 1.9.1 is included in the travis
+    # virtualenv but we want to used numpy installed through apt-get
+    # install.
+    deactivate
+    # Create a new virtualenv using system site packages for numpy and scipy
+    virtualenv --system-site-packages testvenv
+    source testvenv/bin/activate
+    pip install nose
```

Comment:
Yeah virtualenv mis feature.  I have no love for virtualenv. It's an ugly and fragile hack.  I wish the python community didn't consider it as the standard solution. 



On Feb 1, 2015, 17:45, at 17:45, "Loïc Estève" <notifications@github.com> wrote:
>> @@ -47,7 +47,15 @@ if [[ "$DISTRIB" == "conda" ]]; then
>>  
>>  elif [[ "$DISTRIB" == "ubuntu" ]]; then
>>      # Use standard ubuntu packages in their default version
>> -    sudo apt-get install -qq python-scipy python-nose python-pip
>> +    sudo apt-get install -qq python-scipy
>> +    # At the time of writing numpy 1.9.1 is included in the travis
>> +    # virtualenv but we want to used numpy installed through apt-get
>> +    # install.
>> +    deactivate
>> +    # Create a new virtualenv using system site packages for numpy
>and scipy
>> +    virtualenv --system-site-packages testvenv
>> +    source testvenv/bin/activate
>> +    pip install nose
>
>That's something I tried at one point and that failed for nilearn. The
>only reason I could fathom: the system nosetests ends up running as if
>you were outside the virtualenv. 
>
>Note that using system nose should work for scikit-learn because you
>don't install any dependencies in the virtualenv (for nilearn nibabel
>is installed in the virtualenv). That does seem very appealing though.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/scikit-learn/scikit-learn/pull/4193/files#r23898909